patches/api/0480-Item-stack-name-tooltip-API.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: radsteve <radsteve@radsteve.net>
+Date: Sun, 7 Jul 2024 10:51:46 +0200
+Subject: [PATCH] Item stack name tooltip API
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 7c56182acaf827f4b1a986a61cea8e9960604c98..5899dcdcbb95c2d3da3869ca202270ce7b106336 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -3855,6 +3855,21 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     boolean isChunkSent(long chunkKey);
+     // Paper end
+ 
++    // Paper start - Item stack name tooltip API
++    /**
++     * Checks whether the currently held item is displaying the item name tooltip on the client.
++     * However this behavior can be changed via the Accessibility Settings, which can lead to inaccuracies.
++     * @return true if it is showing with default settings, false if not
++     */
++    boolean isItemNameTooltipShown();
```

Comment:
I am a bit iffy on the method name.
As you correctly highlighted, this method may be completely inaccurate if the client changes their notificationDisplayTime.

Idk if we can get the fact that this method is more a guess/approximation into the method name.